CRM/Core/BAO/CustomField.php

```diff
@@ -1073,6 +1073,19 @@ private static function formatDisplayValue($value, $field, $entityId = NULL) {
         }
         else {
           $display = CRM_Utils_Array::value($value, $field['options'], '');
+          // For float type (see Number and Money) $value would be decimal like
+          // 1.00 (because it is stored in db as decimal), while options array
+          // key would be integer like 1. In this case expression on line above
+          // would return empty string (default value), despite the fact that
+          // matching option exists in the array.
+          // In such cases we could just get intval($value) and fetch matching
+          // option again, but this would not work if key is float like 5.6.
+          // So we need to truncate trailing zeros to make it work as expected.
+          if ($display === '' && strpos($value, '.') !== FALSE) {
+            // Truncate trailing zeros, e.g: 5.00 -> 5, 5.60 -> 5.6.
+            $value = rtrim($value, '.0');
```

Comment:
(1) Worth noting:

```php
echo rtrim("123.4", ".0");    # ===> 123.4, ok
echo rtrim("411.0", ".0");    # ===> 411, ok
echo rtrim("5", ".0");        # ===> 5, ok
echo rtrim("50", ".0");       # ===> 5, uh oh
echo rtrim("500.0", ".0");    # ===> 5, uh oh
```

Maybe `preg_replace()` does what you want?

(2) It sounds like the underlying problem is that `CRM_Utils_Array::value()` uses the wrong match-operator for decimal-ish data. I think you can resolve by either normalizing the data (*untested pseudocode*):

```php
$normalize = function($number) { return sprintf("%.5f", $number); };
$normalOptions = CRM_Utils_Array::rekey($field['options'], $normalize);
$display = CRM_Utils_Array::value($normalize($value), $normalOptions, '');
```

or explicitly using a decimal-friendly comparator (*untested psuedocode*):

```php
$comparator = function($a,$b) { return sprintf("%.5f", $a) == sprintf("%.5f", $b); });
// Or: $comparator = function($a,b) {return abs($a -$b) < TOLERANCE; /*ex: 0.001*/ }

$display = keyMatch($value, $field['options'], $comparator, '');

// Pick the $keyValuePair which matches $targetKey (as determined by the $comparator).
function keyMatch($targetKey, $keyValuePairs, $comparator, $default = NULL) {
  foreach ($keyValuePairs as $candidateKey => $candidateValue) {
    if ($comparator($candidateKey, $targetKey)) return $candidateValue;
  }
  return $default;
}
```